pkg/user/user.go

```diff
@@ -178,3 +179,15 @@ func verifyPassword(toVerify string, encoded []byte) (bool, error) {
 
 	return ok, nil
 }
+
+func (u *User) PasswordReset() *PasswordReset {
+	return u.passwordReset
+}
+
+func (u *User) CreatePasswordReset() {
+	u.passwordReset = NewPasswordReset()
+}
+
+func (u *User) RemovePasswordReset() {
+	u.passwordReset = nil
+}
```

Comment:
```suggestion
func (u *User) SetPasswordReset(pr *PasswordReset) {
	u.passwordReset = pr.Clone()
}
```

This implementation is one of answer.

My way is using setter that simplifies User struct because the internal concerns of password reset can be completely transferred to PasswordReset struct. Copying is required to prevent changing internal values by outside without methods.